app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.kt

```diff
@@ -128,11 +133,45 @@ class ZimHostActivity : BaseActivity(), ZimHostCallbacks, ZimHostContract.View {
   private fun startStopServer() {
     when {
       ServerUtils.isServerStarted -> stopServer()
-      selectedBooksPath.size > 0 -> startHotspotManuallyDialog()
+      selectedBooksPath.size > 0 -> {
+        /*val wifiManager =
+          applicationContext.getSystemService(Context.WIFI_SERVICE) as WifiManager*/
+        when {
+          wifiManagerUtils.checkWifi() -> startWifiDialog()
```

Comment:
are we sure that this is what we want? Wifi is just on, not strictly actually connected to anything

Reply:
I improved the condition such that now the dialog is shown only if wifi is on and is connected to a network.